### PR TITLE
Stop the date picker making API requests for invalid dates and times

### PR DESCRIPTION
**Pull request by rstrange-scottlogic**

### CODE REVIEW

File: air-quality-ui/src/components/header/ForecastBaseDatePicker.spec.tsx [hunk 1]
```diff
@@ -19,14 +19,76 @@ describe('ForecastBaseDatePicker component', () => {
...
+
+      fireEvent.change(datePicker, {
+        target: { value: futureDate.toFormat("dd'/'LL'/'yyyy' 'HH':'mm") },
+      })
+      await waitFor(() => {
+        expect(mockSetForecastBaseDate).not.toHaveBeenCalledWith(futureDate)
+      })
+    },
+  )
+  it.each<number>([2, 3, 4, 5, 6, 7, 8, 9, 10, 11])(
```
Comment (lincent) [inline air-quality-ui/src/components/header/ForecastBaseDatePicker.spec.tsx:75]: is the clock 24 hours?

File: air-quality-ui/src/components/header/ForecastBaseDatePicker.tsx [hunk 2]
```diff
@@ -14,19 +14,29 @@ export const ForecastBaseDatePicker = (): JSX.Element => {
     },
   })
 
+  const IsTimeInvalid = (value: DateTime) => {
+    return value.minute != 0 || (value.hour != 0 && value.hour != 12)
```
Comment (lincent) [inline air-quality-ui/src/components/header/ForecastBaseDatePicker.tsx:18]: missing a superb moment to use remainder
```suggestion
    return value.minute != 0 || value.hour % 12 != 0
```

File: air-quality-ui/src/components/header/ForecastBaseDatePicker.spec.tsx (hunk 1 above)
Comment (rstrange-scottlogic) [reply air-quality-ui/src/components/header/ForecastBaseDatePicker.spec.tsx]: Good point! Have added in the others

File: air-quality-ui/src/components/header/ForecastBaseDatePicker.tsx (hunk 2 above)
Comment (rstrange-scottlogic) [reply air-quality-ui/src/components/header/ForecastBaseDatePicker.tsx]: You're quite right, thanks! 👍

**Review by lincent — APPROVED** [review]

**Pull request merged**